### DISCUSSION TITLE
Missing the option to declare preferred runners

### DISCUSSION THREAD

**Discussion by gerhard** [community/community#30693] [Actions] [answered]
Given two sets of runners, `fast` & `free`,  I would like to be able to specify "prefer `fast`, but schedule on `free` if no `fast` runners available". The added benefit is that if no `fast` runner is online, `free` runners are a good fall-back.

Something as simple as `jobs.<job_id>.runs-on-preferred: [fast, free]` would work for my use-case.

Would others find this useful?

---

GitHub Docs: [`jobs.<job_id>.runs-on`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on)

**Comment by Badisi** [comment, marked as answer]
Hello @gerhard,

To answer your question: yes others found this useful already.

This might help you:
- Priority for self-hosted runners ([#25309](https://github.com/orgs/community/discussions/25309))
- Add ability to prioritize GitHub Action runners ([#1665](https://github.com/actions/runner/issues/1665))

**Reply by gerhard** [reply]
Thank you @Badisi! https://github.com/actions/runner/issues/1665 is what I was looking for.

**Comment by Goku446** [comment]
Ffc
